File: FrontEndReactYosr-integrated/FrontEndReactYosr/services/messageService.ts

import { api } from "./api";
import { getToken } from "../utils/storage";

/* =========================
   RÉCUPÉRER LES MESSAGES
========================= */

export const getConversationMessages = async (
  conversationId: number,
  userId: number
) => {
  const token = await getToken();

  const response = await api.get(
    `/messages/conversation/${conversationId}`,
    {
      params: {
        userId,
      },
      headers: {
        Authorization: `Bearer ${token}`,
      },
    }
  );

  return response.data;
};

/* =========================
   ENVOYER UN MESSAGE
========================= */

export const sendMessage = async (
  conversationId: number,
  senderId: number,
  contenu: string,
  image?: string | null
) => {
  const token = await getToken();

  const response = await api.post(
    "/messages",
    {
      conversationId,
      senderId,
      contenu,
      image: image ?? null,
    },
    {
      headers: {
        Authorization: `Bearer ${token}`,
      },
    }
  );

  return response.data;
};

/* =========================
   MARQUER UNE CONVERSATION
   COMME LUE
========================= */

export const markConversationAsRead = async (
  conversationId: number,
  userId: number
) => {
  const token = await getToken();

  await api.put(
    `/messages/conversation/${conversationId}/read`,
    null,
    {
      params: {
        userId,
      },
      headers: {
        Authorization: `Bearer ${token}`,
      },
    }
  );
};

/* =========================
   MARQUER UN MESSAGE
   COMME LU
========================= */

export const markMessageAsRead = async (
  messageId: number,
  userId: number
) => {
  const token = await getToken();

  await api.put(
    `/messages/${messageId}/read`,
    null,
    {
      params: {
        userId,
      },
      headers: {
        Authorization: `Bearer ${token}`,
      },
    }
  );
};

/* =========================
   SUPPRIMER
========================= */

export const deleteMessage = async (
  messageId: number,
  userId: number
) => {
  const token = await getToken();

  await api.delete(
    `/messages/${messageId}`,
    {
      params: {
        userId,
      },
      headers: {
        Authorization: `Bearer ${token}`,
      },
    }
  );
};

/* =========================
   NON LUS
========================= */

export const countUnreadMessages = async (
  userId: number
) => {
  const token = await getToken();

  const response = await api.get(
    `/messages/unread/${userId}`,
    {
      headers: {
        Authorization: `Bearer ${token}`,
      },
    }
  );

  return response.data;
};

File: backendYosrSpring/src/main/java/org/example/backendyosrmegaapp/Services/MessageService.java
package org.example.backendyosrmegaapp.Services;

import org.example.backendyosrmegaapp.entities.ChatMessageResponse;
import org.springframework.web.multipart.MultipartFile;

import java.io.IOException;
import java.util.List;

public interface MessageService {

    ChatMessageResponse sendMessage(
            Long conversationId,
            Long senderId,
            String contenu,
            MultipartFile image
    ) throws IOException;

    ChatMessageResponse sendTextMessage(
            Long conversationId,
            Long senderId,
            String contenu,
            String image
    );

    List<ChatMessageResponse> getMessages(
            Long conversationId,
            Long userId
    );

    void markMessageAsRead(
            Long messageId,
            Long userId
    );

    void markConversationAsRead(
            Long conversationId,
            Long userId
    );

    long countUnreadMessages(
            Long userId
    );

    void deleteMessage(
            Long messageId,
            Long userId
    );

    long countUnreadConversations(Long userId);
}